project-base/storefront/middleware.ts

```diff
@@ -29,11 +29,6 @@ export const middleware: NextMiddleware = async (request) => {
 
         const { search } = new URL(request.url);
         const queryParams = new URLSearchParams(search);
-        const slugTypeQueryParam = queryParams.get('slugType');
-
-        if (slugTypeQueryParam) {
-            return rewriteDynamicPages(slugTypeQueryParam as FriendlyPageTypesValue, request.url, search);
-        }
 
```

Comment:
Your change essentially makes it so that with every navigation, we inquire about the type of page. This is obviously much more robust (we do not need to specify page types in ExtendedNextLink), but it takes longer (every request, every navigation). Moreover, if we wanted to keep this adjustment, we would end up with a really large amount of redundant code, which is there precisely so that we don’t have to inquire about the page type every time.

Reply:
I understand that now, thank you for an explanation. I have removed the type from `ExtendedNextLink` in `NavigationItem` component. This change triggers the script for the type lookup in middleware (fetch request to `resolve-friendly-url`) only for Navigation links.